src/lib/snack-bar/snack-bar.ts

```diff
@@ -0,0 +1,124 @@
+import {
+  NgModule,
+  ModuleWithProviders,
+  Injectable,
+  ComponentRef,
+} from '@angular/core';
+import {
+  Overlay,
+  OverlayState,
+  OverlayRef,
+  ComponentType,
+  ComponentPortal,
+  OverlayModule,
+  PortalModule,
+  OVERLAY_PROVIDERS,
+} from '@angular2-material/core';
+import {CommonModule} from '@angular/common';
+import {MdSnackBarConfig} from './snack-bar-config';
+import {MdSnackBarRef} from './snack-bar-ref';
+import {MdSnackBarContainer} from './snack-bar-container';
+import {SimpleSnackBar} from './simple-snack-bar';
+
+export {MdSnackBarRef} from './snack-bar-ref';
+export {MdSnackBarConfig} from './snack-bar-config';
+
+// TODO(josephperrott): Animate entrance and exit of snack bars.
+// TODO(josephperrott): Automate dismiss after timeout.
+
+
+/**
+ * Service to dispatch Material Design snack bar messages.
+ */
+@Injectable()
+export class MdSnackBar {
+  /** A reference to the current snack bar in the view. */
+  private _snackBarRef: MdSnackBarRef<any>;
+
+  constructor(private _overlay: Overlay) {}
+
+  /**
+   * Creates and dispatches a snack bar with a custom component for the content, removing any
+   * currently opened snack bars.
+   */
+  openFromComponent<T>(component: ComponentType<T>,
+                       config: MdSnackBarConfig): MdSnackBarRef<T> {
+    if (this._snackBarRef) {
+      this._snackBarRef.dismiss();
+    }
+    let overlayRef = this._createOverlay();
+    let snackBarContainer = this._attachSnackBarContainer(overlayRef, config);
+
+    return this._fillSnackBarContainer(component, snackBarContainer, overlayRef);
+  }
+
+
+  /**
+   * Creates and dispatches a snack bar.
+   */
+  open(message: string, buttonLabel: string,
+       config: MdSnackBarConfig): MdSnackBarRef<SimpleSnackBar> {
```

Comment:
Because the config includes the viewContainerRef, it will always need to be included.